src/EventStore.php

```diff
@@ -359,7 +350,7 @@ public function rollback(): void
 
         $this->actionEventEmitter->dispatch($event);
 
-        $this->recordedEvents = new ArrayIterator();
+        $this->resetRecordedEvents();
```

Comment:
I'm not sure if it is worth it, to create a new method for a one line of code? I would prefer `$this->recorderEvents = new AppendIterator()`. Every function call in PHP is a performance drawback too.